deploy/shakenfist_ci/base.py

```diff
@@ -132,6 +140,33 @@ def _await_event(self, instance_uuid, operation, message=None, after=None):
             'After time %s, instance %s had no event "%s:%s" (waited 5 mins)' % (
                 after, instance_uuid, operation, message))
 
+    def _await_image_download_success(self, url):
+        return self._await_image_event(
+            url, 'fetch', 'success')
+
+    def _await_image_download_error(self, url):
+        return self._await_image_event(
+            url, 'fetch', 'Name or service not known')
+
```

Comment:
The only catch with this approach is it means failing tests always take 300 seconds to timeout, whereas if we asserted that waiting for a success means that there are no failures then we could fail faster?

Reply:
Yes, there is a better way. I sort that out after dinner.